src/components/remote_control/src/resource_allocation_manager_impl.cc

```diff
@@ -200,5 +203,24 @@ void ResourceAllocationManagerImpl::OnDriverDisallowed(
   list_of_rejected_resources.push_back(module_type);
 }
 
-ResourceAllocationManager::~ResourceAllocationManager() {}
+void ResourceAllocationManagerImpl::OnUnregisterApplication(
+    const uint32_t app_id) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  rejected_resources_for_application_.erase(app_id);
+  for (AllocatedResources::const_iterator it = allocated_resources_.begin();
```

Comment:
@dev-gh seems that `std::remove_if` is not suitable for associative containers like `std::map`
More information you can find [here](http://www.cplusplus.com/forum/general/131917/)